tests/functional/codegen/features/test_transient.py

```diff
@@ -55,3 +45,497 @@ def setter(k: address, v: uint256):
+    """
+    values = ("0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", ([1],), [[3, 4], [5, 6]])
+
+    c = get_contract(code)
+    assert c.foo(*values) == values
+    assert c.my_struct() == (None, ([],), [])
+    assert c.foo(*values) == values
+
+
+def test_complex_transient_modifiable(get_contract):
+    code = """
+struct MyStruct:
+    a: uint256
+
+my_struct: public(transient(MyStruct))
+
+@external
+def foo(a: uint256) -> MyStruct:
+    self.my_struct = MyStruct(a=a)
+
+    # struct members are modifiable after initialization
+    self.my_struct.a += 1
+
+    return self.my_struct
+    """
+
+    c = get_contract(code)
+    assert c.foo(1) == (2,)
+    assert c.my_struct() == (0,)
+    assert c.foo(1) == (2,)
+
+
+def test_list_transient(get_contract):
+    code = """
+my_list: public(transient(uint256[3]))
+
+@external
+def foo(_a: uint256, _b: uint256, _c: uint256) -> uint256[3]:
+    self.my_list = [_a, _b, _c]
+    return self.my_list
+    """
+    values = (100, 42, 23230)
+
+    c = get_contract(code)
+    assert c.foo(*values) == list(values)
+    for i in range(3):
+        assert c.my_list(i) == 0
+    assert c.foo(*values) == list(values)
+
+
+def test_hashmap_transient(get_contract):
+    code = """
+my_map: public(transient(HashMap[uint256, uint256]))
+
+@external
+def foo(k: uint256, v: uint256) -> uint256:
+    self.my_map[k] = v
+    return self.my_map[k]
+    """
+    c = get_contract(code)
+    for v in range(5):
+        for k in range(5):
+            assert c.foo(k, v) == v
+            assert c.my_map(k) == 0
+
+
+def test_complex_hashmap_transient(get_contract):
+    code = """
+struct MyStruct:
+    a: uint256
+    b: DynArray[uint256, 3]
+
+my_map: public(transient(HashMap[uint256, MyStruct]))
+my_res: public(HashMap[uint256, MyStruct])
+
+@external
+def do_side_effects():
+    a: DynArray[uint256, 3] = [1, 2, 3]
+    s: MyStruct = MyStruct(a=100, b=a)
+    for i: uint256 in range(2):
+        for j: uint256 in range(3):
+            s.b[j] = i + j
+        s.a = i
+        self.my_map[i] = s
+        self.my_res[i] = self.my_map[i]
+    """
+    c = get_contract(code)
+    c.do_side_effects(transact={})
+    for i in range(2):
+        assert c.my_res(i)[0] == i
+        assert c.my_map(i)[0] == 0
+        for j in range(3):
+            print(c.my_res(i)[1])
+            assert c.my_res(i)[1][j] == i + j
+            assert c.my_map(i)[1] == []
+
+
+def test_dynarray_transient(get_contract):
+    code = """
+my_list: public(transient(DynArray[uint256, 3]))
+
+@external
+def get_my_list(_a: uint256, _b: uint256, _c: uint256) -> DynArray[uint256, 3]:
+    self.my_list = [_a, _b, _c]
+    return self.my_list
+
+@external
+def get_idx_two(_a: uint256, _b: uint256, _c: uint256) -> uint256:
+    self.my_list = [_a, _b, _c]
+    return self.my_list[2]
+    """
+    values = (100, 42, 23230)
+
+    c = get_contract(code)
+    assert c.get_my_list(*values) == list(values)
+    with pytest.raises(TransactionFailed):
```

Comment:
is this on purpose instead of tx_failed()?

Reply:
nope, I was not aware of that! I will change to tx_failed().